simvue/filter.py

```diff
@@ -0,0 +1,231 @@
+        self._filters.append(f"name == {name}")
+        return self
+    
+    def has_name_containing(self, name: str) -> Self:
+        self._filters.append(f"name contains {name}")
+        return self
+    
+    def created_within(self, *, hours: int=0, days: int=0, years: int=0) -> Self:
+        return self._time_within(Time.Created, hours=hours, days=days, years=years)
+    
+    def has_description_containing(self, search_str: str) -> Self:
+        self._filters.append(f"description contains {search_str}")
+        return self
+    
+    def exclude_description_containing(self, search_str: str) -> Self:
+        self._filters.append(f"description not contains {search_str}")
+        return self
+    
+    def has_tag(self, tag: str) -> Self:
+        self._filters.append(f"has tag.{tag}")
+        return self
+
+    def as_list(self) -> list[str]:
+        return self._filters
+
+    def clear(self) -> None:
+        self._filters = []
+
+
+class FoldersFilter(RestAPIFilter):
+    def has_path(self, name: str) -> "FoldersFilter":
+        self._filters.append(f"path == {name}")
+        return self
+    
+    def has_path_containing(self, name: str) -> "FoldersFilter":
+        self._filters.append(f"path contains {name}")
+        return self
+    
+    def _generate_members(self) -> None:
+        return super()._generate_members()
+
+
+class RunsFilter(RestAPIFilter):
+    def _generate_members(self) -> None:
+        _global_comparators = [
+            self._value_contains,
+            self._value_eq,
+            self._value_neq
+        ]
+
+        _numeric_comparators = [
+            self._value_geq,
+            self._value_leq,
+            self._value_lt,
+            self._value_gt
+        ]
+
+        for label, system_spec in System.__members__.items():
+            for function in _global_comparators:
+                _label: str = label.lower()
+                _func_name: str = function.__name__.replace("_value", _label)
+                _out_func = lambda value, func=function: func("system", system_spec.value, value)
+                _out_func.__name__ = _func_name
+                setattr(self, _func_name, _out_func)
+
+        for function in _global_comparators + _numeric_comparators:
+            _func_name: str = function.__name__.replace("_value", "metadata")
+            _out_func = lambda attribute, value, func=function: func("metadata", attribute, value)
+            _out_func.__name__ = _func_name
+            setattr(self, _func_name, _out_func)
+
+    def author(self, username: str="self") -> "RunsFilter":
+        self._filters.append(f"user == {username}")
+        return self
+
+    def exclude_author(self, username: str="self") -> "RunsFilter":
+        self._filters.append(f"user != {username}")
+        return self
+    
+    def starred(self) -> "RunsFilter":
+        self._filters.append("starred")
+        return self
+    
+    def has_name(self, name: str) -> "RunsFilter":
```

Comment:
There's some inconsistency with some of these function names. Eg author and name can both be filtered on but one uses `.author` and the other `.has_name`.

I'm also not sure about switching between `has_` and `is_` in the status filters. Why not just always use `is_` for these?

Personally I'd probably go for complete consistency across the filters by dispensing with the prefix verbs altogether (ie have `.status`, `.running`, `.lost`, `.alert`, `.name_contains`, `.author_not` (or `.not_author`), `.no_metadata_attribute`, `.start_within`, `.folder`, etc), like with `.starred`. But that's not necessary.